### PR TITLE
Bug 1616242 - provide FE test coverage for server side filters

### PR DESCRIPTION
**Pull request by airimovici**
[Bug link](https://bugzilla.mozilla.org/show_bug.cgi?id=1616242)

### CODE REVIEW

File: tests/ui/perfherder/alerts_test.jsx [hunk 1]
```diff
@@ -76,14 +76,21 @@ const mockModifyAlert = {
...
+        hideAssignedToOthers: undefined,
+        filter: undefined,
+        updateParams,
+      }}
       user={testUser}
       projects={repos}
       location={{
         pathname: '/alerts',
-        search: '',
+        search: id ? 'id=20174' : '',
```
Comment (octavian-negru) [inline tests/ui/perfherder/alerts_test.jsx]: I think `id=20174` it's a little bit too magical.
Why this?

Comment (airimovici) [reply tests/ui/perfherder/alerts_test.jsx]: I forgot it there while doing some tests. I'll remove it

File: tests/ui/perfherder/alerts_test.jsx [hunk 2]
```diff
@@ -219,6 +227,81 @@ test('clicking the star icon for an alert updates that alert', async () => {
...
+
+  fireEvent.click(allFromDropdown[1]);
+  expect(updateParamsMock).toHaveBeenCalledTimes(2);
+  expect(updateParamsMock.mock.calls[1][0]).toEqual({
+    framework: -1,
+  });
+});
+
+test('filters are server side when in List Mode', async () => {
+  expect(true).toBe(true);
```
Comment (ionutgoldan) [inline tests/ui/perfherder/alerts_test.jsx]: I believe these 2 tests aren't done.

Comment (octavian-negru) [reply tests/ui/perfherder/alerts_test.jsx]: Not yet, work in progress.

Comment (airimovici) [reply tests/ui/perfherder/alerts_test.jsx]: I removed the empty tests

**Comment by airimovici** [comment]
@camd can you take a look on this PR? The are some unit tests I did some time ago and today I brought them up to date.

**Comment by sarah-clements** [comment]
@airimovici It looks like this was approved a while back but needs conflicts to be resolved before merging. If you aren't going to do that anytime soon, please close this pr.

**Pull request closed without merging**